kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerObservable.scala

```diff
@@ -214,12 +221,125 @@ object KafkaConsumerObservable {
     */
   def apply[K, V](cfg: KafkaConsumerConfig, topicsRegex: Regex)(
     implicit K: Deserializer[K],
-    V: Deserializer[V]): KafkaConsumerObservable[K, V] = {
+    V: Deserializer[V]): KafkaConsumerObservable[K, V, ConsumerRecord[K, V]] = {
 
     val consumer = createConsumer[K, V](cfg, topicsRegex)
     apply(cfg, consumer)
   }
 
+  /**
+    * Builds a [[KafkaConsumerObservable]] instance with ability to manual commit offsets
+    * and forcibly disables auto commits in configuration.
+    * Such instances emit [[CommittableMessage]] instead of Kafka's ConsumerRecord.
+    *
+    * Usage example:
+    * {{{
+    *   KafkaConsumerObservable.manualCommit[String,String](consumerCfg, List(topicName))
+    *     .map(message => message.record.value() -> message.committableOffset)
+    *     .mapTask { case (value, offset) => performBusinessLogic(value).map(_ => offset) }
+    *     .bufferTimedAndCounted(1.second, 1000)
+    *     .mapTask(offsets => CommittableOffsetBatch(offsets).commit())
+    *     .subscribe()
+    * }}}
+    *
+    * @param cfg is the [[KafkaConsumerConfig]] needed for initializing the
+    *        consumer; also make sure to see `monix/kafka/default.conf` for
+    *        the default values being used. Auto commit will disabled and
+    *        observable commit order will turned to [[monix.kafka.config.ObservableCommitOrder.NoAck NoAck]] forcibly!
+    *
+    * @param consumer is a factory for the
+    *        `org.apache.kafka.clients.consumer.KafkaConsumer`
+    *        instance to use for consuming from Kafka
+    * */
+  def manualCommit[K, V](
+    cfg: KafkaConsumerConfig,
+    consumer: Task[KafkaConsumer[K, V]]): KafkaConsumerObservable[K, V, CommittableMessage[K, V]] = {
+
+    def consumerCommitBatch(consumer: KafkaConsumer[K, V], batch: Map[TopicPartition, Long]): Unit =
```

Comment:
I wonder whether we should/could just expose `commitSync` an `commitAsync` (with possibility to specify callback) instead of taking it from config. What do you think?

Reply:
I made it similar to the existing function:
```scala
  def consumerCommit(consumer: KafkaConsumer[K, V]): Unit =
      config.observableCommitType match {
        case ObservableCommitType.Sync =>
          blocking(consumer.commitSync())
        case ObservableCommitType.Async =>
          blocking(consumer.commitAsync())
      }
```

I support your idea about callback for `commitAsync`. Do you want to expose it in `manualCommit` parameters?
```scala

def manualCommitSync[K, V](
    cfg: KafkaConsumerConfig,
    consumer: Task[KafkaConsumer[K, V]]): KafkaConsumerObservable[K, V, CommittableMessage[K, V]] = ???

def manualCommitAsync[K, V](
    cfg: KafkaConsumerConfig,
    consumer: Task[KafkaConsumer[K, V]],
    callback: OffsetCommitCallback): KafkaConsumerObservable[K, V, CommittableMessage[K, V]] = ???

```

Reply:
I was thinking about having two `commit` variants in `CommittableOffset` so users can choose which version they want any time. Do you think it makes sense to allow it?

Reply:
It is really very convenient. Great idea!